### PR TITLE
Struct init type hints

### PR DESCRIPTION
**Pull request by nullptrdevs**
Partial #1531

Techatrix you get to name the config option :)

### CODE REVIEW

File: src/features/inlay_hints.zig [hunk 1]
```diff
@@ -450,6 +450,35 @@ fn writeNodeInlayHint(
...
+        .struct_init_dot_two_comma,
+        .struct_init_dot,
+        .struct_init_dot_comma,
+        .struct_init,
+        .struct_init_comma,
+        => {
+            var buffer: [2]Ast.Node.Index = undefined;
+            const struct_init = tree.fullStructInit(&buffer, node) orelse return;
+            for (struct_init.ast.fields) |value_node| { // the node of `value` in `.name = value`
+                const name_token = tree.firstToken(value_node) - 2; // get --------------^^^^
```
Comment (nullptrdevs) [inline src/features/inlay_hints.zig:465]: Note to self: Disable inlay hints before aligning comments ...

**Comment by Techatrix** [comment]
> Techatrix you get to name the config option :)

Coming up with a name for a config options is definitely not my strong suit. (╥﹏╥)
I would suggest something like "inlay_hints_show_anonymous_literal_field_type" or "inlay_hints_show_anon_literal_field_type"?

**Comment by nullptrdevs** [comment]
Are they anon literals though? + it works for regular `S{.name<T> = ..}` too

**Comment by Techatrix** [comment]
> Are they anon literals though? + it works for regular `S{.name<T> = ..}` too

oh right, then "inlay_hints_show_struct_literal_field_type"?

File: src/features/inlay_hints.zig [hunk 2]
```diff
@@ -462,6 +462,36 @@ fn writeNodeInlayHint(
...
+        .struct_init_dot_two,
+        .struct_init_dot_two_comma,
+        .struct_init_dot,
+        .struct_init_dot_comma,
+        .struct_init,
+        .struct_init_comma,
+        => {
+            if (!builder.config.inlay_hints_show_anon_literal_field_type) return;
+            var buffer: [2]Ast.Node.Index = undefined;
+            const struct_init = tree.fullStructInit(&buffer, node) orelse return;
```
Comment (Techatrix) [inline src/features/inlay_hints.zig:476]:
```suggestion
            const struct_init = tree.fullStructInit(&buffer, node).?;
```

File: src/features/inlay_hints.zig [hunk 3]
```diff
@@ -462,6 +462,36 @@ fn writeNodeInlayHint(
...
+                const field = decl.handle.tree.fullContainerField(decl.decl.ast_node) orelse return;
+                if (field.ast.type_expr == 0) return;
+                try appendTypeHintString(
+                    builder,
+                    name_token,
+                    try reduceTypeWhitespace(
+                        offsets.nodeToSlice(decl.handle.tree, field.ast.type_expr),
+                        builder.arena,
+                    ),
+                );
```
Comment (Techatrix) [inline src/features/inlay_hints.zig:492]:
```suggestion
                const decl = (try builder.analyser.getSymbolEnumLiteral(builder.arena, builder.handle, name_loc.start, name)) orelse continue;
                const ty = try decl.resolveType(builder.analyser) orelse continue;
                const type_str: []const u8 = try std.fmt.allocPrint(builder.arena, "{}", .{ty.fmt(builder.analyser)});
                if (type_str.len == 0) continue;
                try appendTypeHintString(
                    builder,
                    name_token,
                    try reduceTypeWhitespace(type_str, builder.arena),
                );
```

**Review by Techatrix — CHANGES_REQUESTED** [review]

**Review by Techatrix — APPROVED** [review]

**Pull request merged**